test/queues.test.ts

```diff
@@ -0,0 +1,143 @@
+/*!
+ * This source file is part of the EdgeDB open source project.
+ *
+ * Copyright 2019-present MagicStack Inc. and the EdgeDB authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {LifoQueue} from "../src/queues";
+
+test("LifoQueue `get` awaits for items in the queue", async (done) => {
+  let result = -1;
+  const queue = new LifoQueue<number>();
+
+  queue.get().then((value: number) => {
+    expect(value).toBe(result);
+
+    done();
```

Comment:
Hm... can we rewrite this in async/await terms? I don't understand if this `done()` is here by mistake and should be moved to the below `setTimeout` closure, or if it's here on purpose.

Reply:
Done here is right, because `get()` gets resolved after an item is pushed to the queue. I am rewriting to use async/await syntax.